test/apps/iodemo/io_demo.cc

```diff
@@ -472,9 +484,11 @@ class DemoClient : public P2pDemoCommon {
         size_t data_size = get_data_size();
 
         if (!send_io_message(conn, IO_WRITE, sn, data_size)) {
-            return data_size;
+            return 0;
         }
 
+        int index = get_conn_index(conn);
```

Comment:
- pass server_info_t* to do_io_read/do_io_write instead of UcxConnection* - it will allow counting reads/writes
- for err handling case, save std::map<conn*,int> to find the connection id